### PR TITLE
Add Foreign Key and Join Queries 🔢

### PR DESCRIPTION
**Pull request by gisachris**
## IN This PR I present the Update, Delete And Aggregate Queries:👨‍💻:

Below are some of my implementations as well as going over the project requirements 🤝 .
## Requirements
- use GitFlow 🌊.
- Add to the `schema.SQL` file with the required content. ⛱️
> Create an `owners` Table.
owners_table_creation
> Create a `Species` Table.
species_table_creation
> Modify `animals` table
     ‣Make sure that id is set as autoincremented PRIMARY KEY
     ‣Remove column species
drop_species_original
     ‣Add column `species_id` which is a foreign key referencing `species` table 
add_species_id_animals
     ‣Add column `owner_id` which is a foreign key referencing the `owners` table
add_owners_id_animals

- Add to the `data.SQL` file with the required content. 🛰️
> Insert  Data into the `owners` table
insert_owners
> Insert  Data into the `species` table
insert_species
> Modify your inserted `animals` so it includes the `species_id` value
      ‣If the name ends in "mon" it will be Digimon
update_species_id_mon
      ‣All other animals are Pokemon
update_species_id_non_mon
> Modify your inserted animals to include owner information (owner_id):
      ‣Sam Smith owns Agumon.
      ‣Jennifer Orwell owns Gabumon and Pikachu.
      ‣Bob owns Devimon and Plantmon. 
      ‣Melody Pond owns Charmander, Squirtle, and Blossom.
      ‣Dean Winchester owns Angemon and Boarmon.
update_give_owners_animals

- Add to the `query.SQL` file with the required content. 🧭
> Write queries (using `JOIN`) to answer the following questions :
      ‣What animals belong to Melody Pond?
melody_pond_animals
      ‣List of all animals that are `pokemon` (their type is `Pokemon`).
all_pokemon_species
      ‣List all owners and their animals, remember to include those that don't own any animal.
all_owners_all animals
      ‣How many animals are there per species?
grouped_species_amounts
      ‣List all Digimon owned by Jennifer Orwell.
digimon_orwell
      ‣List all animals owned by Dean Winchester that haven't tried to escape.
escape_attempts_dean
      ‣Who owns the most animals?
owners_with_most_animals

## Achievements
- use GitFlow 🌊 .✅
- Add to the `schema.SQL` file with the required content. ⛱️✅
- Add to the `data.SQL` file with the required content. 🛰️✅
- Add to the `query.SQL` file with the required content. 🧭✅

## Conclusion
**We are looking forward to hearing 👂  from you Our CR in terms of what can be better implemented 👨‍💻  for the project thankyou!. And we look forward to hearing from you soon.**  🙇

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -5,5 +5,17 @@ CREATE TABLE animals(
     escape_attempts INT,
     neutered BOOLEAN,
     weight_kg DECIMAL(18,2),
-    species VARCHAR
+    species_id INT FOREIGN KEY,
+    owners_id INT FOREIGN KEY
+);
```
Comment (Whoistolu) [inline schema.sql]: - [ ] [REQUIRED] Kindly note that the project requirement states that you should modify the animals table. I can see that you just inserted the column `species_id` and the column `owner_id` into an existing table which is not correct. To modify the `animals` table, you need to use the ALTER TABLE command. This command will help you modify an existing table. Take a look at [this](https://www.w3schools.com/sql/sql_alter.asp) for more insight.

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @gisachris 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (gisachris) [reply schema.sql]: Thank you @Whoistolu  for the great Review and the suggested documentation. I did add it before but I will make it more clear. 🙏

**Comment by gisachris** [comment]
## IN This PR I present the Requested ♻️ Changes Of the Foreign Key and Join Queries: 📦

Below are some of my implementations as well as going over the Required changes 🤝 .
## Requirements
- Correctly alter the Animals 🦁 table to add project requirements [see here](https://github.com/gisachris/vet-clinic/pull/3/commits/ba14841254c92881a86ed2835ef9ac5b2f6066f4)

## Achievements
- Correctly alter the Animals 🦁 table to add project requirements ✅

## Conclusion
**We are looking forward to hearing 👂  from you Our CR in terms of what can be better implemented 👨‍💻  for the project thankyou!. And we look forward to hearing from you soon.**  🙇

**Review by abdulhamiid — APPROVED** [review]
# :green_circle:  Approved! :partying_face:
Hi @gisachris 👋, 
image
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🌟

- ✔️ Query multiple tables
- ✔️ Use primary key & foreign key mechanisms for joining tables.

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**